hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/AbstractSqmAttributeJoin.java

```diff
@@ -91,11 +93,11 @@ public void clearFetched() {
 	}
 
 	private void validateFetchAlias(String alias) {
-		if ( fetchJoin && alias != null && nodeBuilder().isJpaQueryComplianceEnabled() ) {
-			throw new IllegalStateException(
-					"The JPA specification does not permit specifying an alias for fetch joins."
-			);
-		}
+//		if ( fetchJoin && alias != null && nodeBuilder().isJpaQueryComplianceEnabled() ) {
```

Comment:
This check should probably be skipped if we know that the alias is a generated one. We will need a naming scheme for generated aliases anyway since we need to warn users not to use such names in HQL to avoid collisions when creating `CriteriaQuery` from HQL.

Reply:
I wonder if  it's possible to make `generateAlias()` avoid collisions automatically.